src/file_pipe_log/reader.rs

```diff
@@ -64,17 +65,49 @@ impl<F: FileSystem> LogItemBatchFileReader<F> {
     /// Returns the next [`LogItemBatch`] in current opened file. Returns
     /// `None` if there is no more data or no opened file.
     pub fn next(&mut self) -> Result<Option<LogItemBatch>> {
-        if self.valid_offset < self.size {
+        // @lucasliang.
+        // TODO: [Fulfilled in writing progress when DIO is open.]
+        // We should also consider that there might exists broken blocks when DIO
+        // is open, and the following reading strategy should tolerate reading broken
+        // blocks until it finds an accessible header of `LogBatch`.
+        let fs_block_size = get_system_block_size();
+        while self.valid_offset < self.size {
+            debug_assert!(self.file_context.is_some());
+            let data_layout = self.file_context.as_ref().unwrap().format.data_layout();
             if self.valid_offset < LOG_BATCH_HEADER_LEN {
                 return Err(Error::Corruption(
                     "attempt to read file with broken header".to_owned(),
                 ));
             }
-            let (footer_offset, compression_type, len) = LogBatch::decode_header(&mut self.peek(
-                self.valid_offset,
-                LOG_BATCH_HEADER_LEN,
-                0,
-            )?)?;
+            let (footer_offset, compression_type, len) = {
+                match LogBatch::decode_header(&mut self.peek(
+                    self.valid_offset,
+                    match data_layout {
+                        DataLayout::NoAlignment => LOG_BATCH_HEADER_LEN,
+                        DataLayout::Alignment => {
+                            // In DataLayout::Alignment mode, tail data in this block maybe
+                            // contain an valid header of the adjacent LogBatch or paddings
+                            // with `0`.
+                            round_up(self.valid_offset, fs_block_size) - self.valid_offset
+                                + LOG_BATCH_HEADER_LEN
+                        }
+                    },
+                    0,
+                )?) {
+                    Err(e) => {
+                        // In DataLayout::Alignment mode, tail data in the previous block may be
```

Comment:
Check it is indeed padding by looking at the first and last u64.

Reply:
I think its no need for it. Because what we are concerned are valid headers of LogBatch, whatever the padding is filled with, we just need to fetch and get the next valid aligned offset, followed with parsing. So, we can just skip the rest bytes by `round_up(self.valid_offset, fs_block_size as usize)`.

Reply:
You can't just skip data whenever you want. They may contain user data that is corrupted by disk failure.